pkg/volume/host_path/host_path.go

```diff
@@ -103,6 +104,12 @@ func (plugin *hostPathPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volum
 	if err != nil {
 		return nil, err
 	}
+
+	err = validation.ValidatePathNoBacksteps(hostPathVolumeSource.Path)
```

Comment:
SetUp is the function that has the mounter actually act on the configuration... the only reason it's a no-op for hostpath is that no setup or checks are currently needed or performed. If you consider a plugin like nfs, if it was given an invalid hostname, it wouldn't know until it tried to perform the mount and failed in SetUp.